planner/core/planbuilder.go

```diff
@@ -396,17 +396,20 @@ func (b *PlanBuilder) buildSelectLock(src LogicalPlan, lock ast.SelectLockType)
 	return selectLock
 }
 
-func (b *PlanBuilder) buildPrepare(x *ast.PrepareStmt) Plan {
+func (b *PlanBuilder) buildPrepare(x *ast.PrepareStmt) (Plan, error) {
 	p := &Prepare{
 		Name: x.Name,
 	}
 	if x.SQLVar != nil {
-		// TODO: Prepared statement from variable expression do not work as expected.
-		// p.SQLText, _ = x.SQLVar.GetValue().(string)
+		if v, ok := b.ctx.GetSessionVars().Users[x.SQLVar.Name]; ok {
+			p.SQLText = v
+		} else {
+			p.SQLText = "NULL"
+		}
 	} else {
 		p.SQLText = x.SQLText
 	}
-	return p
+	return p, nil
```

Comment:
Why adding one more return value error which is always nil?

Reply:
I think we may address some errors in this function first thought; but found no such errors should be address here.

Reply:
Then I prefer removing it.

Reply:
Address this comment makes an LGTM. @AndrewDi